bundles/org.openhab.binding.solarmax/src/main/java/org/openhab/binding/solarmax/internal/connector/SolarMaxConnector.java

```diff
@@ -0,0 +1,396 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solarmax.internal.connector;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * 
+ * The {@link SolarMaxConnector} class is used to communicated with the SolarMax device (on a binary level)
+ *
+ * With a little help from https://github.com/sushiguru/solar-pv/blob/master/solmax/pv.php
+ * 
+ * @author Jamie Townsend - Initial contribution
+ */
+public class SolarMaxConnector {
+
+    /**
+     * default port number of SolarMax devices is...
+     */
+    final private static int DEFAULT_PORT = 12345;
+
+    private static final Logger logger = LoggerFactory.getLogger(SolarMaxConnector.class);
+
+    /**
+     * default timeout for socket connections is 1 second
+     */
+    private static int connectionTimeout = 1000;
```

Comment:
This can be `final`. Constants shall be all upper case.

Reply:
final-ised & capitalised